src/main/java/com/example/bookshelf/aspect/AspectMeasureExecutionMethods.java

```
@@ -0,0 +1,41 @@
+package com.example.bookshelf.aspect;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+@Component
+@Aspect
+public class AspectMeasureExecutionMethods {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Pointcut("within(com.example.bookshelf.service.*) && within(com.example.bookshelf.exception.*)")
+    private void allMethodsService() {
+    }
+
+    @Around("allMethodsService()")
+    public Object watchTime(ProceedingJoinPoint joinPoint) {
+
+        long start = System.currentTimeMillis();
+        Object output = null;
+        String args = Arrays.stream(joinPoint.getArgs())
+                .map(arg -> arg.toString())
+                .collect(Collectors.joining(","));
+        try {
+            output = joinPoint.proceed();
+        } catch (Throwable throwable) {
+            throwable.printStackTrace();
```

never do prinnt stack trace in production code, use the corresponding logger method

---

moreover, you hiding possible exception here and it will lead to hidden errors that will not report to the user